hook.go

```diff
@@ -69,22 +88,15 @@ func (h *Hook) Patch(spec *ebpf.CollectionSpec, hookMapSymbol string) error {
 		return nil
 	}
 
-	collectionAbi := ebpf.CollectionABI{
-		Maps: map[string]*ebpf.MapABI{
-			hookMapSymbol: &HookMapABI,
-		},
-	}
-
-	err := collectionAbi.CheckSpec(spec)
```

Comment:
This used to check the ABI of the map as well, I think we should add a call to `CheckHook()` to preserve that behaviour.

Reply:
I didn't find map ABI checks in the `CheckSpec` method; can you provide a link to that check please?

Reply:
Ah yes, you're right, that's `CollectionABI.Check()`, `CollectionABI.CheckSpec()` doesn't do that.

Reply:
So, I think only https://github.com/cloudflare/xdpcap/pull/31#discussion_r347086489 remains unresolved.
I'll be grateful if you specify the parts which I should move to internal package.
Thank you.

Reply:
Sorry, got side tracked. Thanks for following up on this! :)